diff --git a/zephyr/drivers/spi/spi_gecko.c b/zephyr/drivers/spi/spi_gecko.c
index d07485793b..4d59d6185d 100644
--- a/zephyr/drivers/spi/spi_gecko.c
+++ b/zephyr/drivers/spi/spi_gecko.c
@@ -33,7 +33,9 @@ LOG_MODULE_REGISTER(spi_gecko);
 #define CLOCK_USART(id) _CONCAT(cmuClock_USART, id)
 #define GET_GECKO_USART_CLOCK(n) CLOCK_USART(DT_INST_PROP(n, peripheral_id))
 #else
-#if (USART_COUNT <= 2)
+#if (USART_COUNT <= 1)
+#define CLOCK_USART(ref) 	cmuClock_USART0
+#elif (USART_COUNT <= 2)
 #define CLOCK_USART(ref)	(((ref) == USART0) ? cmuClock_USART0 \
 			       : ((ref) == USART1) ? cmuClock_USART1 \
 			       : -1)
