diff --git a/zephyr/drivers/display/ssd1306.c b/zephyr/drivers/display/ssd1306.c
index f8e8e3c..0f17293 100644
--- a/zephyr/drivers/display/ssd1306.c
+++ b/zephyr/drivers/display/ssd1306.c
@@ -193,8 +193,8 @@ static inline int ssd1306_set_charge_pump(const struct device *dev)
 {
        const struct ssd1306_config *config = dev->config;
        uint8_t cmd_buf[] = {
-               (config->sh1106_compatible ? SSD1306_SET_CHARGE_PUMP_ON : SSD1306_SET_CHARGE_PUMP_ON),
-               (config->sh1106_compatible ? SSD1306_SET_CHARGE_PUMP_ON_ENABLED
+               (config->sh1106_compatible ? SH1106_SET_DCDC_MODE : SSD1306_SET_CHARGE_PUMP_ON),
+               (config->sh1106_compatible ? SH1106_SET_DCDC_ENABLED
                                           : SSD1306_SET_CHARGE_PUMP_ON_ENABLED),
                SSD1306_PANEL_PUMP_VOLTAGE,
        };
